src/app/components/shell/header/main-menu/login-menu/login-menu.js

```diff
@@ -1,170 +1,116 @@
 import settings from 'settings';
-import componentType from '~/helpers/controller/init-mixin';
-import busMixin from '~/helpers/controller/bus-mixin';
-import Dropdown from '../dropdown/dropdown';
-import {description as template} from './login-menu.html';
+import React, {useState, useEffect} from 'react';
+import {useToggle} from '~/components/jsx-helpers/jsx-helpers.jsx';
+import {useUserModel} from '~/models/usermodel';
 import userModelBus from '~/models/usermodel-bus';
 import linkHelper from '~/helpers/link';
+import Dropdown, {MenuItem} from '../dropdown/dropdown';
 
-const spec = {
-    template,
-    view: {
-        classes: ['nav-menu-item', 'rightmost']
-    },
-    model() {
-        return {
-            loggedIn: this.loggedIn,
-            loginUrl: linkHelper.loginLink()
+function useLocation() {
+    const [location, setLocation] = useState(window.location.href);
+
+    useEffect(() => {
+        const updateLocation = () => {
+            setLocation(window.location.href);
         };
-    }
-};
+
+        window.addEventListener('navigate', updateLocation);
+        return () => window.removeEventListener('navigate', updateLocation);
+    }, []);
+
+    return location;
+}
 
```

Comment:
will useLocation be needed elsewhere?  maybe move to a separate file if so?

Reply:
Seems likely. Moving.